atomic_reactor/plugins/exit_pulp_publish.py

```diff
@@ -0,0 +1,123 @@
+"""
+Copyright (c) 2017 Red Hat, Inc
+All rights reserved.
+
+This software may be modified and distributed under the terms
+of the BSD license. See the LICENSE file for details.
+
+Push built image to crane or delete them if the build failed
+
+Several authentication schemes are possible, including
+username+password and key/certificate via secrets.
+
+However, the recommended scheme (since Origin 1.0.6) is to store a
+key and certificate in a secret which the builder service account is
+allowed to mount:
+
+$ oc secrets new pulp ./pulp.key ./pulp.cer
+secrets/pulp
+$ oc secrets add serviceaccount/builder secret/pulp --for=mount
+
+In the BuildConfig for atomic-reactor, specify the secret in the
+strategy's 'secrets' array, specifying a mount path:
+
+"secrets": [{
+  "secretSource": {
+    "name": "pulp"
+  },
+  "mountPath": "/var/run/secrets/pulp"
+}]
+
+In the configuration for this plugin, specify the same path for
+pulp_secret_path:
+
+"pulp_publish": {
+  "pulp_registry_name": ...,
+  ...
+  "pulp_secret_path": "/var/run/secrets/pulp"
+}
+"""
+
+from __future__ import print_function, unicode_literals
+
+from atomic_reactor.constants import PLUGIN_PULP_PUBLISH_KEY
+from atomic_reactor.plugins.build_orchestrate_build import get_worker_build_info
+from atomic_reactor.plugin import ExitPlugin
+from atomic_reactor.util import ImageName
+from atomic_reactor.pulp_util import PulpHandler
+
+
+class PulpPublishPlugin(ExitPlugin):
+    key = PLUGIN_PULP_PUBLISH_KEY
+    is_allowed_to_fail = False
+
+    def __init__(self, tasker, workflow, pulp_registry_name,
+                 pulp_secret_path=None, username=None, password=None,
+                 dockpulp_loglevel=None):
+        """
+        constructor
+
+        :param tasker: DockerTasker instance
+        :param workflow: DockerBuildWorkflow instance
+        :param pulp_registry_name: str, name of pulp registry to use, specified in /etc/
+                                   dockpulp.conf
+        :param pulp_secret_path: path to pulp.cer and pulp.key; $SOURCE_SECRET_PATH otherwise
+        :param username: pulp username, used in preference to certificate and key
+        :param password: pulp password, used in preference to certificate and key
+        """
+        # call parent constructor
+        super(PulpPublishPlugin, self).__init__(tasker, workflow)
+        self.pulp_registry_name = pulp_registry_name
+        self.pulp_secret_path = pulp_secret_path
+        self.username = username
+        self.password = password
+        self.dockpulp_loglevel = dockpulp_loglevel
```

Comment:
Nit: don't need to set these 5 instance variables as they are not used after this method.